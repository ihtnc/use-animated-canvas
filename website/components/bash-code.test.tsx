import { describe, test, expect, vi, beforeEach, afterAll } from 'vitest'
import { render } from '@testing-library/react'
import CodeSnippet from './code-snippet'
import bash from 'highlight.js/lib/languages/bash'
import BashCode from './bash-code'

vi.mock('@/components/code-snippet', async (importOriginal) => ({
  ...await importOriginal<typeof import('@/components/code-snippet')>(),
  default: vi.fn().mockReturnValue(<div id='CodeSnippet' />)
}))

describe('BashCode component', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  afterAll(() => {
    vi.restoreAllMocks()
  })

  test('should render CodeSnippet component', () => {
    const code = 'npm run install'
    const { container } = render(<BashCode code={code} />)

    expect(container).toMatchSnapshot()
  })

  test('should call internal CodeSnippet component', () => {
    const code = 'npm run install'
    const collapsible = false
    render(<BashCode code={code} collapsible={collapsible} />)

    expect(CodeSnippet).toHaveBeenCalledWith({
      code,
      language: 'bash',
      languageFn: bash,
      collapsible
    }, expect.anything())
  })
})