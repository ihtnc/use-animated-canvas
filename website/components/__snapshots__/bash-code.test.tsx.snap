// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BashCode component > should render CodeSnippet component 1`] = `
<div>
  <div
    id="CodeSnippet"
  />
</div>
`;
